enumerables.rb

```diff
@@ -0,0 +1,93 @@
+# frozen_string_literal: true
+
+require './helpers.rb'
+
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield(self[i], i)
+      i += 1
+    end
+  end
+
+  def my_select
+    new_arr = []
+    my_each { |n| new_arr.push(n) if yield(n) }
+    new_arr
+  end
+
+  def my_all?(pattern = nil)
+    my_each { |n| return false unless n.to_s.match(pattern) } if pattern
+    my_each { |n| return false unless yield(n) } unless pattern
+    true
+  end
+
+  def my_any?(pattern = nil)
+    my_each { |n| return true if n.to_s.match(pattern) } if pattern
+    my_each { |n| return true if yield(n) } unless pattern
+    false
```

Comment:
- [ ] my_any? when no block or argument is given returns true if at least one of the collection is not false or nil
- [ ] my_any? when a class is passed as an argument returns true if at least one of the collection is a member of such class